emails/tests/models_tests.py

```diff
@@ -12,29 +12,93 @@
 from model_bakery import baker
 
 from ..models import (
+    address_hash,
     CannotMakeAddressException,
+    CannotMakeSubdomainException,
     DeletedAddress,
     DomainAddress,
+    has_bad_words,
+    NOT_PREMIUM_USER_ERR_MSG,
+    Profile,
     RelayAddress,
-    Profile
+    TRY_DIFFERENT_VALUE_ERR_MSG,
 )
 
 
+class MiscEmailModelsTest(TestCase):
+    def test_has_bad_words_with_bad_words(self):
+        assert has_bad_words('angry')
+
+    def test_has_bad_words_without_bad_words(self):
+        assert not has_bad_words('happy')
+
+    def test_address_hash_without_subdomain(self):
+        address = 'aaaaaaaaa'
+        expected_hash = sha256(f'{address}'.encode('utf-8')).hexdigest()
+        assert address_hash(address) == expected_hash
+    
+    def test_address_hash_with_subdomain(self):
+        address = 'aaaaaaaaa'
+        subdomain = 'test'
+        expected_hash = sha256(f'{address}@{subdomain}'.encode('utf-8')).hexdigest()
+        assert address_hash(address, subdomain) == expected_hash
+
+
 class RelayAddressTest(TestCase):
     def setUp(self):
-        self.user = baker.make(User)
+        user = baker.make(User)
+        self.user_profile = Profile.objects.get(user=user)
 
     def test_make_relay_address_assigns_to_user(self):
-        relay_address = RelayAddress.make_relay_address(self.user)
-        assert relay_address.user == self.user
+        relay_address = RelayAddress.make_relay_address(self.user_profile)
+        assert relay_address.user == self.user_profile.user
 
     def test_make_relay_address_makes_different_addresses(self):
+        user = baker.make(User)
+        random_sub = random.choice(
+            settings.SUBSCRIPTIONS_WITH_UNLIMITED.split(',')
+        )
+        socialaccount = baker.make(
+            SocialAccount,
+            user=user,
+            provider='fxa',
+            extra_data={'subscriptions': [random_sub]}
+        )
+        user_profile = Profile.objects.get(user=user)
         for i in range(1000):
-            RelayAddress.make_relay_address(self.user)
+            RelayAddress.make_relay_address(user_profile)
         # check that the address is unique (deeper assertion that the generated aliases are unique)
```

Comment:
https://github.com/mozilla/fx-private-relay/pull/812/files/f666157d04e7f21ec7054efda0c62fd0e19b6b66#diff-aafaa09204bebf8342a8eb2d04a39ce7a84d25a3bf5dd4779747e22bc7a3b98bR71 I forgot to covert the `relay_addresses` as a`set` rather than a `list` to verify that the addresses are unique. Good catch.

Reply:
1d96e127fb6caf2da80e071448d0ed9f67463e0d